packages/medusa/src/api/routes/admin/product-categories/create-product-category.ts

```diff
@@ -0,0 +1,119 @@
+import { IsNotEmpty, IsOptional, IsString, IsBoolean } from "class-validator"
+import { Request, Response } from "express"
+import { EntityManager } from "typeorm"
+
+import { ProductCategoryService } from "../../../../services"
+import { AdminProductCategoriesReqBase } from "../../../../types/product-category"
+
+/**
+ * @oas [post] /product-categories
+ * operationId: "PostProductCategories"
+ * summary: "Create a Product Category"
+ * description: "Creates a Product Category."
+ * x-authenticated: true
+ * requestBody:
+ *   content:
+ *     application/json:
+ *       schema:
+ *         $ref: "#/components/schemas/AdminPostProductCategoriesReq"
+ * x-codeSamples:
+ *   - lang: JavaScript
+ *     label: JS Client
+ *     source: |
+ *       import Medusa from "@medusajs/medusa-js"
+ *       const medusa = new Medusa({ baseUrl: MEDUSA_BACKEND_URL, maxRetries: 3 })
+ *       // must be previously logged in or use api token
+ *       medusa.admin.productCategories.create({
+ *         name: 'Jeans'
+ *       })
+ *       .then(({ productCategory }) => {
+ *         console.log(productCategory.id);
+ *       });
+ *   - lang: Shell
+ *     label: cURL
+ *     source: |
+ *       curl --location --request POST 'https://medusa-url.com/admin/product-categories' \
+ *       --header 'Authorization: Bearer {api_token}' \
+ *       --header 'Content-Type: application/json' \
+ *       --data-raw '{
+ *           "name": "Jeans"
+ *       }'
+ * security:
+ *   - api_token: []
+ *   - cookie_auth: []
+ * tags:
+ *   - Product Category
+ * responses:
+ *  "200":
+ *    description: OK
+ *    content:
+ *      application/json:
+ *        schema:
+ *          type: object
+ *          properties:
+ *            productCategory:
+ *              $ref: "#/components/schemas/ProductCategory"
+ *  "400":
+ *    $ref: "#/components/responses/400_error"
+ *  "401":
+ *    $ref: "#/components/responses/unauthorized"
+ *  "404":
+ *    $ref: "#/components/responses/not_found_error"
+ *  "409":
+ *    $ref: "#/components/responses/invalid_state_error"
+ *  "422":
+ *    $ref: "#/components/responses/invalid_request_error"
+ *  "500":
+ *    $ref: "#/components/responses/500_error"
+ */
+export default async (req: Request, res: Response) => {
+  const { validatedBody } = req as {
+    validatedBody: AdminPostProductCategoriesReq
+  }
+
+  const productCategoryService: ProductCategoryService = req.scope.resolve(
+    "productCategoryService"
+  )
+
+  const manager: EntityManager = req.scope.resolve("manager")
+  const created = await manager.transaction(async (transactionManager) => {
+    return await productCategoryService
+      .withTransaction(transactionManager)
+      .create(validatedBody)
+  })
+
+  const productCategory = await productCategoryService.retrieve(created.id)
```

Comment:
todo: We should also use the transformQuery middleware and use the retrieveConfig here. So that a user can create and provides the fields and expand for the returned result. It means that the oas need some adjustments as well.